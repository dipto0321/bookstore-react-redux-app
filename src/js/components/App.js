import React from 'react';
import { Route, Router, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import decode from 'jwt-decode';
import BooksDashboard from './BooksDashboard';
import '../../scss/main.scss';
import LoginPage from './LoginPage';
import SignUpPage from './SignUpPage';
import configureStore from '../store/configureStore';
import { setCurrentUser } from '../actions/user';
import { setAuthorizationToken } from '../services/api';
import history from '../services/history';
import withAuth from '../hocs/withAuth';

const store = configureStore();

if (localStorage.token) {
  const userData = decode(localStorage.token);
  setAuthorizationToken(localStorage.token);

  try {
    store.dispatch(setCurrentUser({ authenticated: true, data: userData }));
  } catch (err) {
    store.dispatch(setCurrentUser({ authenticated: false, data: null }));
    localStorage.clear();
  }
}

const App = () => (
  <Provider store={store}>
    <Router history={history}>
      <Switch>
        <Route path="/login" component={withAuth(LoginPage, true)} />
        <Route path="/signup" component={withAuth(SignUpPage, true)} />
        <Route path="/" component={withAuth(BooksDashboard)} />
      </Switch>
    </Router>
  </Provider>
);

export default App;
