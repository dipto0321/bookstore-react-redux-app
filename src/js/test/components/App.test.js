import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App';

describe('App component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
