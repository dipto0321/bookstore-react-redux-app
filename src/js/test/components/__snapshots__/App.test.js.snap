// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Authenticate)",
            "type": [Function],
          }
        }
        path="/login"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Authenticate)",
            "type": [Function],
          }
        }
        path="/signup"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Authenticate)",
            "type": [Function],
          }
        }
        path="/"
      />
    </Switch>
  </Router>
</Provider>
`;
